File: src/main/java/ru/stnk/RestTestAPI/repository/UserRepository.java
package ru.stnk.RestTestAPI.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import ru.stnk.RestTestAPI.entity.User;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {

    Optional<User> findByEmail (String email);
}

File: src/main/kotlin/ru/stnk/resttestapi/repository/UserRepository.kt
package ru.stnk.resttestapi.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import ru.stnk.resttestapi.entity.User
import java.util.*

@Repository
interface UserRepository : JpaRepository<User, Long> {

    fun findByEmail(email: String): Optional<User>

    /*@Query(value = "SELECT * FROM USERS WHERE EMAIL = ?1", nativeQuery = true)
    User findByUserEmail (String email);*/
}
